### PR TITLE
fix: rune popup for v11.18. use lockfile to locate port & token.

### PR DESCRIPTION
**Pull request by magicwenli**
test ok

### CODE REVIEW

File: src/share/file.ts [hunk 1]
```diff
@@ -81,39 +80,23 @@ export const getLatestLogFile = async (dir: string) => {
...
-      console.error(`[LCU] cannot find target renderer log.`)
-      return [null, null, null]
-    }
-
-    const url = content.match(/https(.*)\/index\.html/)?.[1] ?? ``;
-    const token = url.match(/riot:(.*)@/)?.[1] ?? null;
-    const port = url.match(/:(\d+)$/)?.[1] ?? null;
+    const lockfile = await fs.readFile(`${lockfilePath}`, 'utf8');
+    const port = lockfile.split(`:`)[2];
+    const passwd = lockfile.split(`:`)[3];
```
Comment (cangzhang) [inline src/share/file.ts:91]: I think it's better to rename the variable `passwd` to `token`, due to it's more a token than a password

File: src/share/file.ts [hunk 2]
```diff
@@ -81,39 +80,23 @@ export const getLatestLogFile = async (dir: string) => {
   }
 };
 
-const authReg = /https:\/\/riot:.+@127\.0\.0\.1:\d+\/index.html/
 
 export const getLcuToken = async (dirPath: string) => {
   const appendGameToDir = config.get(`appendGameToDir`);
-  const dir = `${appendGameToDir ? `${dirPath}/Game` : dirPath}/Logs/LeagueClient Logs`;
+  const lockfilePath = `${appendGameToDir ? `${dirPath}/LeagueClient` : dirPath}/lockfile`;
```
Comment (cangzhang) [inline src/share/file.ts:86]: Coz lcu of different servers have different folder structure, I will verify it later for NA lcu.

**Review by cangzhang — COMMENTED** [review]
Good work! 👍

**Comment by magicwenli** [comment]
cool

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

File: src/share/file.ts (hunk 1 above)
Comment (magicwenli) [reply src/share/file.ts]: cool

File: src/share/file.ts [hunk 3]
```diff
@@ -81,39 +80,23 @@ export const getLatestLogFile = async (dir: string) => {
...
-    }
-
-    const url = content.match(/https(.*)\/index\.html/)?.[1] ?? ``;
-    const token = url.match(/riot:(.*)@/)?.[1] ?? null;
-    const port = url.match(/:(\d+)$/)?.[1] ?? null;
+    const lockfile = await fs.readFile(`${lockfilePath}`, 'utf8');
+    const port = lockfile.split(`:`)[2];
+    const token = lockfile.split(`:`)[3];
+
+    const url = `://riot:${passwd}@127.0.0.1:${port}`
```
Comment (cangzhang) [inline src/share/file.ts:93]: Looks like this one was forgot. https://github.com/cangzhang/champ-r/runs/3552301364

Comment (magicwenli) [reply src/share/file.ts]: thanks for reviewing 🥵

File: src/share/file.ts (hunk 1 above)
Comment (Sokwva) [reply src/share/file.ts]: good!

**Comment by magicwenli** [comment]
I have an idea for a new feature, adding an automatic acceptance switch. But I am a newbie to Github, I wrote the function in the change/auto-accept branch under magicwenli/champ-r. Now all changes to my main branch will be reflected in this pull request, which is unexpected. 

I don’t know how to merge the latest changes in my repo without affecting this pull request.

btw, do you like this idea?

**Comment by cangzhang** [comment]
well you have to close this PR and make commits in your own repo.

**Comment by cangzhang** [comment]
I will cherry pick your commits and make a new PR coz I have to release a new version later tonight. Try ideas in your own repo and PR is always welcome🤓

**Comment by magicwenli** [comment]
appreciate that! 🥰

**Pull request closed without merging**